web-ui/src/main/resources/catalog/lib/ace/snippets/gn.js

```diff
@@ -0,0 +1,22 @@
+define("ace/snippets/gn", ["require", "exports", "module"], function (e, t, n) {
+  "use strict";
+  t.snippetText =
+    'snippet agg-term\n	"${1:indexfield}":{"terms": {"field": "${1:indexfield}"}}\n' +
+    'snippet agg-term-with-size\n	"${1:indexfield}": {"terms": {"field": "${1:indexfield}", "size": 5}}\n' +
+    'snippet agg-term-with-size-and-filter\n	"${1:indexfield}": {"terms": {"field": "${1:indexfield}", "size": 5, "include": ".*"}, "meta": {"caseInsensitiveInclude": true}}\n' +
+    'snippet agg-histogram\n	"${1:indexfield}": {"histogram": {"field": "${1:indexfield}", "interval": ${2:interval}, "keyed" : true, "min_doc_count": 1}}\n' +
+    'snippet agg-meta\n	"meta": {"collapsed": true, "userHasRole": "isReviewerOrMore"}\n' +
+    '',
+    t.scope = "json"
+});
+(function () {
+  window.require(["ace/snippets/gn"], function (m) {
+    if (typeof module == "object" && typeof exports == "object" && module) {
+      module.exports = m;
+    }
+  });
+})();
```

Comment:
can't that be put in a separate, actual text file? This is hard to read/comprehend

Reply:
I tried, but not a JS expert so has we don't really have a JS build env here it is probably not that easy ...